lib/test/unit/code-snippet-fetcher.rb

```diff
@@ -31,7 +31,11 @@ def read_source(path)
           break if first_line.nil?
           encoding = detect_encoding(first_line) || Encoding::UTF_8
           first_line.force_encoding(encoding)
-          file.set_encoding(encoding)
+          if Gem::Version.create(RUBY_VERSION) >= Gem::Version.create('2.0.0')
+            file.set_encoding(encoding, encoding)
+          else
+            file.set_encoding(encoding)
+          end
```

Comment:
How about stop using `set_encoding` entirely?

```diff
diff --git a/lib/test/unit/code-snippet-fetcher.rb b/lib/test/unit/code-snippet-fetcher.rb
index 81f946d..00a561b 100644
--- a/lib/test/unit/code-snippet-fetcher.rb
+++ b/lib/test/unit/code-snippet-fetcher.rb
@@ -31,9 +31,11 @@ module Test
           break if first_line.nil?
           encoding = detect_encoding(first_line) || Encoding::UTF_8
           first_line.force_encoding(encoding)
-          file.set_encoding(encoding)
           lines << first_line
-          lines.concat(file.readlines)
+          file.each_line do |line|
+            line.force_encoding(encoding)
+            lines << line
+          end
         end
         lines
       end
```

Reply:
It looks better than the original patch! fixed.  https://github.com/test-unit/test-unit/pull/166/commits/906e0d464a16bb858b2006ec79a91535709541b2